File: packages/api/configs/cors.go
package configs

import (
	"fmt"
	"strings"

	"github.com/gofiber/fiber/v2"
	"github.com/gofiber/fiber/v2/middleware/cors"
)

func GenerateCORSConfig() func(*fiber.Ctx) error {
	config := cors.ConfigDefault

	config.AllowOrigins = GenerateOrigins()
	return cors.New(config)
}

func GenerateOrigins() string {
	ALLOWED_LOCALHOST_PROTOCOLS := []string{"http", "https"}
	ALLOWED_LOCALHOST_DOMAINS := []string{"localhost", "127.0.0.1"}
	ALLOWED_LOCALHOST_PORTS := []string{"3000", "3001", "6969"}
	ALLOWED_PUBLIC_DOMAINS := []string{
		"https://dev.thehub-aubg.com",
		"https://thehub-aubg.com",
	}

	var origins []string

	for _, protocol := range ALLOWED_LOCALHOST_PROTOCOLS {
		for _, domain := range ALLOWED_LOCALHOST_DOMAINS {
			for _, port := range ALLOWED_LOCALHOST_PORTS {
				origin := fmt.Sprintf("%s://%s:%s", protocol, domain, port)
				origins = append(origins, origin)
			}
		}
	}

	origins = append(origins, ALLOWED_PUBLIC_DOMAINS...)

	return strings.Join(origins, ",")
}

File: packages/py-api/py_api/utilities/cors.py


from logging import getLogger
from typing import Final, List

logger = getLogger("CORS")


def construct_origins() -> List[str]:
    ALLOWED_LOCALHOST_PORTS: Final = [3000, 3001]
    ALLOWED_LOCALHOST_DOMAINS: Final = ["localhost", "127.0.0.1"]
    ALLOWED_LOCALHOST_PROTOCOLS: Final = ["http", "https"]
    origins = [f"{protocol}://{domain}:{port}" for protocol in ALLOWED_LOCALHOST_PROTOCOLS for domain in ALLOWED_LOCALHOST_DOMAINS for port in ALLOWED_LOCALHOST_PORTS]

    # there's no need for ports here since the api is behind a proxy
    ALLOWED_PUBLIC_DOMAINS = [
        "https://dev.thehub-aubg.com",
        "https://thehub-aubg.com",
    ]

    origins.extend(ALLOWED_PUBLIC_DOMAINS)

    return origins
